src/sage/groups/perm_gps/permgroup.py

```diff
@@ -1581,6 +1581,83 @@ def smallest_moved_point(self):
         p = self._libgap_().SmallestMovedPoint()
         return self._domain_from_gap[Integer(p)]
 
+    @cached_method
+    def disjoint_direct_product_decomposition(self):
+        r"""
+        Returns the finest partition of the underlying set such that ``self``
+        is isomorphic to the direct product of the projections of ``self``
+        onto each part of the partition. Each part is a union of orbits
+        of ``self``.
+
+        The algorithm is from [CJ2022]_, which runs in time polynomial in
+        `n \cdot |G|`, where `n` is the degree of the group and `|G|` is
```

Comment:
`G` usually denotes a group and `|G|` usually denotes the size of the group, so I'd suggest to use another letter, for example, `X` for the generating set.